src/modules/cpu/kernel/tensor_max.hpp

```diff
@@ -0,0 +1,825 @@
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+RppStatus tensor_max_u8_u8_host(Rpp8u *srcPtr,
+                                RpptDescPtr srcDescPtr,
+                                Rpp8u *maxArr,
+                                Rpp32u maxArrLength,
+                                RpptROIPtr roiTensorPtrSrc,
+                                RpptRoiType roiType,
+                                RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(srcDescPtr->n)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp8u *srcPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp8u *srcPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+
+        Rpp32u alignedLength = (bufferLength / 96) * 96;
+        Rpp32u vectorIncrement = 96;
+        Rpp32u vectorIncrementPerChannel = 32;
+
+        // Tensor max 1 channel (NCHW)
+        if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+        {
+            alignedLength = (bufferLength / vectorIncrementPerChannel) * vectorIncrementPerChannel;
+            vectorIncrement = vectorIncrementPerChannel;
+            Rpp8u max = 0;
+            Rpp8u resultAvx[16];
+
+            Rpp8u *srcPtrRow;
+            srcPtrRow = srcPtrChannel;
+#if __AVX2__
+                __m256i pMax = _mm256_setzero_si256();
+#endif
+                for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+                {
+                    Rpp8u *srcPtrTemp;
+                    srcPtrTemp = srcPtrRow;
+
+                    int vectorLoopCount = 0;
+#if __AVX2__
+                    for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                    {
+                        __m256i p1;
+                        rpp_simd_load(rpp_load32_u8_avx, srcPtrTemp, &p1);
+                        compute_max_32_host(&p1, &pMax);
+
+                        srcPtrTemp += vectorIncrement;
+                    }
+#endif
+                    for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                    {
+                        max = std::max(*srcPtrTemp++, max);
+                    }
+                    srcPtrRow += srcDescPtr->strides.hStride;
+                }
+#if __AVX2__
```

Comment:
I don't know what this is doing outside the iterative loop. We need to review this code

Reply:
This part of code is used to finally narrow down max/min to 1 final value for PLN1 and 4 values(min/max R, min/max G, min/max B, min/max all channels).